### PR TITLE
samba: update to version 4.12.5

### PR DESCRIPTION
**Pull request by chincheta0815**

### CODE REVIEW

**Review by AndWac — APPROVED** [review]
Builds fine on my build server. I haven't tested the result, though.

File: components/network/samba/patches/15-util_paths_c_TEMP_FOR_4.12.patch [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ 		if (szPath == NULL) {
+-			return NULL;
++			goto done;
+ 		}
+-		len_written = snprintf(buf, sizeof(buf), "%s", szPath);
+-		if (len_written >= sizeof(buf) || len_written < 0) {
+-			/* Output was truncated or an error. */
++		len = strnlen(szPath, PATH_MAX);
++		if (len >= PATH_MAX) {
+ 			return NULL;
```
Comment (pyhalov) [inline components/network/samba/patches/15-util_paths_c_TEMP_FOR_4.12.patch]: shouldn't we free buf here?

**Comment by chincheta0815** [comment]
I simply backported this patch from the source. I did not code anything on my oen.

**Comment by pyhalov** [comment]
Reported https://bugzilla.samba.org/show_bug.cgi?id=14441 . I'd fixed patch and merged this.

**Comment by pyhalov** [comment]
@chincheta0815 Can you fix the patch - https://github.com/samba-team/samba/commit/aa4d135710e0fdcf92f522b3c6228717105c0775 ?

**Comment by chincheta0815** [comment]
I will look into it today and think I will get it. ;o)

**Comment by chincheta0815** [comment]
@pyhalov I hope I got it correctly. There was just one of the lines mentioned. The other "NULL" seemed okay. BUT remark: I am not a good programmer/memfree'er.

**Pull request merged**